Document '_name' field in 'function_score' query's function definition (#7340)
opensearch-project/documentation-website: _query-dsl/compound/function-score.md

  }
}
```
</details>

## Named functions

When defining a function, you can specify its name using the `_name` parameter at the top level. This name is useful for debugging and understanding the scoring process. Once specified, the function name is included in the score calculation explanation whenever possible (this applies to functions, filters, and queries). You can identify the function by its `_name` in the response.

### Example

> The following request sets `explain` to `true` for debugging purposes in order to obtain scoring explanation in the response. Each function contains a `_name` parameter so you can identify the function unambiguously:

Reviewer: ```suggestion
The following request sets `explain` to `true` for debugging purposes in order to obtain a scoring explanation in the response. Each function contains a `_name` parameter so that you can identify the function unambiguously:
```

                      "description": "product of:",
                      "details": [
                        {
                          "value": 0.35040614,
                          "description": "Function for field comments:",
                          "details": [
                            {
                              "value": 0.35040614,
                              "description": "exp(-0.5*pow(MIN[Math.max(Math.abs(16.0(=doc value) - 1000.0(=origin))) - 0.0(=offset), 0)],2.0)/461662.4130844683, _name: comments_function)",
                              "details": []
                            }
                          ]
                        },
                        {
                          "value": 0.1,
                          "description": "weight",
                          "details": []
                        }
                      ]
                    }
                  ]
                },
                {
                  "value": 3.4028235e+38,
                  "description": "maxBoost",
                  "details": []
                }
              ]
            }
          ]
        }
      }
    ]
  }
}
```
</details>

> As you can see, the explanation contains the name in the respective function's description.

Reviewer: ```suggestion
```

> As you can see, the explanation contains the name in the respective function's description.

Reviewer: Same information already provided above.

  }
}
```
</details>

## Named functions

> You can specify `_name` parameter for every function top level definition. This parameter is used to identify the function in the response. The name is useful for debugging and understanding the scoring process. Once specified, it is returned back as a part of the explanation whenever it dims possible (it applies to functions, filters and queries).

Reviewer 2: ```suggestion
When defining a function, you can specify its name using the `_name` parameter at the top level. This name is useful for debugging and understanding the scoring process. Once specified, the function name is included in the score calculation explanation whenever possible (this applies to functions, filters, and queries). You can identify the function by its `_name` in the response.
```

### Example

> The following request sets `explain: true` for debugging purpose to retrieve the explanation of the scoring process in the response. For the ease of identifying unambiguously the functions, the `_name` parameter is set for each function:

Reviewer 2: ```suggestion
The following request sets `explain` to `true` for debugging purposes in order to obtain scoring explanation in the response. Each function contains a `_name` parameter so you can identify the function unambiguously:
```

```json
GET blogs/_search
{
  "explain": true,
  "size": 1,
  "query": {
    "function_score": {
      "functions": [
        {
          "_name": "likes_function",
          "script_score": {
            "script": {
              "lang": "painless",
              "source": "return doc['likes'].value * 2;"
            }
          },
          "weight": 0.6
        },
        {
          "_name": "views_function",
          "field_value_factor": {
            "field": "views",
            "factor": 1.5,
            "modifier": "log1p",
            "missing": 1
          },
          "weight": 0.3
        },
        {
          "_name": "comments_function",
          "gauss": {
            "comments": {
              "origin": 1000,
              "scale": 800
            }
          },
          "weight": 0.1
        }
      ]
    }
  }
}
```
{% include copy-curl.html %}

> The response will contain the explanation of the scoring process with the names of the functions in it:

Reviewer 2: ```suggestion
The response explains the scoring process. For each function, the explanation contains the function `_name` in its `description`:
```

                      "description": "product of:",
                      "details": [
                        {
                          "value": 0.35040614,
                          "description": "Function for field comments:",
                          "details": [
                            {
                              "value": 0.35040614,
                              "description": "exp(-0.5*pow(MIN[Math.max(Math.abs(16.0(=doc value) - 1000.0(=origin))) - 0.0(=offset), 0)],2.0)/461662.4130844683, _name: comments_function)",
                              "details": []
                            }
                          ]
                        },
                        {
                          "value": 0.1,
                          "description": "weight",
                          "details": []
                        }
                      ]
                    }
                  ]
                },
                {
                  "value": 3.4028235e+38,
                  "description": "maxBoost",
                  "details": []
                }
              ]
            }
          ]
        }
      }
    ]
  }
}
```
</details>

> {% include copy-curl.html %}

Reviewer 2: ```suggestion
```

              "lang": "painless",
              "source": "return doc['likes'].value >= 800;"
            },
            "weight": 0.7
          }
        },
        {
          "field_value_factor": {
            "_name": "views_function",
            "field": "views",
            "factor": 1.5,
            "modifier": "log1p",
            "missing": 1
          },
          "weight": 0.3
        }
      ]
    }
  }
}
```
{% include copy-curl.html %}

The response will contain the explanation of the scoring process with the names of the functions in it:

<details open markdown="block">
  <summary>
    Response
  </summary>
  {: .text-delta}

```json
{
  "took": 14,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 3,
      "relation": "eq"
    },
    "max_score": 31.191923,
    "hits": [
      {
        "_index": "blogs",
        "_id": "3",
        "_score": 31.191923,
        "_source": {
          "name": "Distributed tracing with Data Prepper",
          "views": 800,
          "likes": 50,
          "comments": 5,
          "date_posted": "2022-04-25"
        },

>         "_explanation": {}

Author: ⏳ Here I'm working on setting up an analogical index and fetch actual real OpenSearch response's explanation 👍🏻
